### ISSUE TITLE
The Icons look very blurry on Retina / High Resolution Displays

### ISSUE THREAD

**Issue by mustii82** [mgarin/weblaf#455]
We must use Icons in Higher Resolution to make the Components look better on Retina and High Resolution Devices.

**Comment by mgarin** [comment]
This is unfortunately a general issue with Swing UI - it doesn't properly scale on high-DPI screens as it was not planned to be supported in Swing from the beginning.

There have been some adjustments made since JDK6, but they haven't really improved situation a lot. You might see two possible cases with different JDK being used for Swing UI on high-DPI screens:

1. UI is natively upscaled making it blurry (mostly with newer JDK versions)
2. UI is crisp but extremely small (mostly with older JDK versions)

You are probably seeing the first case which means the whole Swing UI that is painted on the buffer image is upscaled and blurred to be more or less usable on high-DPI retina display. That means that all elements including icons are also upscaled and changing their source quality won't really matter as the upscaling happens on a raster image.

I do plan to try out some major UI scaling options, but it will surely not come in the next update. You can check #352 where related discussions have happened before.

So long story short - I'm limited by JDK and Swing options here and unfortunately cannot provide a quick solution, but hopefully can implement my own proper UI scaling with later updates.

**Comment by mgarin** [comment]
I'll be closing this issue, but we can continue scaling-related discussion in #352.
